titus-supplementary-component/tasks-publisher/src/main/java/com/netflix/titus/supplementary/taskspublisher/TitusClientImpl.java

```diff
@@ -182,4 +169,25 @@ private void configureMetrics(Registry registry) {
         return serviceStub.withInterceptors(MetadataUtils.newAttachHeadersInterceptor(metadata));
     }
 
+    private void buildCacheForJobs() {
+        jobs = Caffeine.newBuilder()
+                .maximumSize(MAX_CACHE_SIZE)
+                .buildAsync(new AsyncCacheLoader<String, Job>() {
+                    @Nonnull
+                    @Override
+                    public CompletableFuture<Job> asyncLoad(@Nonnull String jobId, @Nonnull Executor executor) {
+                        CompletableFuture<Job> jobResult = new CompletableFuture<>();
+                        ListenableFuture<Job> jobFuture = jobManagementServiceFutureStub.findJob(JobId.newBuilder().setId(jobId).build());
+                        jobFuture.addListener(() -> {
```

Comment:
probably need to do more here, e.g.: https://github.com/spotify/futures-extra/blob/master/src/main/java/com/spotify/futures/ListenableToCompletableFutureWrapper.java